### PR TITLE
Add docs on substepping field solver

### PR DESCRIPTION
**Pull request by sbastrakov**
Clarify its properties and add a suggested use case.

### CODE REVIEW

**Review by psychocoderHPC — DISMISSED** [review]

File: docs/source/models/AOFDTD.rst [hunk 1]
```diff
@@ -434,6 +434,30 @@ Usage
...
 
+Substepping
+-----------
+Any field solver can be combined with substepping in time.
+In this case, each iteration of the main PIC loop involves multiple invocations of the chosen field solver.
+Substepping is fully compatible with other numerics, such as absorbers, incident field, laser generation.
+A substepping field solver has the same orders of accuracy in spatial and time steps as the base solver.
+
+A user sets main PIC time step value as usual in :ref:`grid.param <usage-params-core>`, and selects the number of field solver substeps via template argument `T_numSubsteps`.
+Field solver will internally operate with :math:`\Delta t_{sub} = \Delta t / T_numSubsteps`.
```
Comment (steindev) [inline docs/source/models/AOFDTD.rst]: 'T_numSubsteps' -> 'T_{numSubsteps}'

File: docs/source/models/AOFDTD.rst [hunk 2]
```diff
@@ -434,6 +434,30 @@ Usage
...
+Solver properties including the Courant-Friedrichs-Lewy condition are then expressed with :math:`\Delta t_{sub}`, which is less restricting.
+However note that regardless of field solver and substepping PIConGPU also limits its main time step to
```
Comment (steindev) [inline docs/source/models/AOFDTD.rst]: Missing comma. Maybe just 'However, regardless of field solver...'

File: docs/source/models/AOFDTD.rst [hunk 3]
```diff
@@ -434,6 +434,30 @@ Usage
...
+
+.. math::
+
+   \Delta t_{max} < \frac{1}{c} \min\{ \Delta x, \Delta y, \Delta z \}
+
+for 3D; for 2D the requirement is similar but does not involve :math:`\Delta z`.
+
+Still, with substepping solver a user could sometimes increase the value of :math:`\Delta t` used.
+Consider a simplified case of :math:`\Delta x = \Delta y = \Delta z`, and Yee solver with a time step value near the  Courant-Friedrichs-Lewy condition threshold :math:`\Delta t_{base} = 0.995 \frac{\Delta x}{c \sqrt 3}.
```
Comment (steindev) [inline docs/source/models/AOFDTD.rst]: Missing backtic '`' at end of sentence.

File: docs/source/models/AOFDTD.rst [hunk 4]
```diff
@@ -434,6 +434,30 @@ Usage
...
+Raising the main time step to `t_{max}` and substepping by 2 would make the field solver have :math:`\Delta t_{sub} < \Delta t_{base}` thus satisfying the Courant-Friedrichs-Lewy condition and providing slightly better resolution.
+Whether this approach is applicable then depends on whether `t_{max}` sufficiently resolves plasma wavelength and other time scales in the rest of the PIC loop.
```
Comment (steindev) [inline docs/source/models/AOFDTD.rst]: missing ':math:' before '`t_{max}`'

File: docs/source/models/AOFDTD.rst [hunk 5]
```diff
@@ -434,6 +434,30 @@ Usage
...
+
+.. math::
+
+   \Delta t_{max} < \frac{1}{c} \min\{ \Delta x, \Delta y, \Delta z \}
+
+for 3D; for 2D the requirement is similar but does not involve :math:`\Delta z`.
+
+Still, with substepping solver a user could sometimes increase the value of :math:`\Delta t` used.
+Consider a simplified case of :math:`\Delta x = \Delta y = \Delta z`, and Yee solver with a time step value near the  Courant-Friedrichs-Lewy condition threshold :math:`\Delta t_{base} = 0.995 \frac{\Delta x}{c \sqrt 3}.
+Raising the main time step to `t_{max}` and substepping by 2 would make the field solver have :math:`\Delta t_{sub} < \Delta t_{base}` thus satisfying the Courant-Friedrichs-Lewy condition and providing slightly better resolution.
```
Comment (steindev) [inline docs/source/models/AOFDTD.rst]: missing ':math:' before '`t_{max}`'

**Review by steindev — CHANGES_REQUESTED** [review]
I am fine with the content, just a few TeX related comments.

File: docs/source/models/AOFDTD.rst (hunk 1 above)
Comment (sbastrakov) [reply docs/source/models/AOFDTD.rst]: It's the name of the template parameter of that solver.

**Comment by sbastrakov** [comment]
@steindev thanks for your suggestions, they are implemented.

**Review by steindev — APPROVED** [review]

**Pull request merged**